src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -132,7 +140,21 @@ public String getScript() {
         if (script.isEmpty()) {
             listener.getLogger().println("Warning: was asked to run an empty script");
         }
-        AgentInfo agentInfo = ws.act(new getAgentInfo());
+
+        final Jenkins jenkins = Jenkins.get();
+
+        Manifest manifest;
+        try (InputStream manifestStream = DurableTask.class.getResourceAsStream("/META-INF/MANIFEST.MF")) {
+            manifest = new Manifest(manifestStream);
+        }
+
+        Node wsNode = jenkins.getNode(FilePathUtils.getNodeName(ws));
+        FilePath nodeRoot = wsNode.getRootPath();
```

Comment:
So I noticed that workspace is always two levels below root of the node, is this always the case?
i.e., do I even need to bother getting the node to get the root? Instead, I can  just  call  `getParent` on the workspace twice. I do that  already in the  unit  tests

Reply:
I'm not sure. I would add a new agent to Jenkins, open all of the advanced config and read the help text, but what you have seems safer. I also don't know if there is a preferred place to cache things on agents. There is a passing mention of agent caches [here](https://github.com/jenkinsci/jenkins/blob/a5a92af4bebc73298d2061e8e95227d4e416fc74/core/src/main/resources/hudson/model/Slave/help-remoteFS.html), but that's all I found at a glance.

Reply:
Added a new agent and  dug around, but only found the same thing  you found.

Reply:
Hmm, I just realized that `FilePathUtils` comes from workflow-api, so you had to add a new dependency to get this to work. I don't think we want to add a dependency on a Pipeline-specific plugin here. Can you use `ws.toComputer().getNode()` instead? I think you'd need some null checks that fall back to not using caching given the limitations of `FilePath.toComputer`, mentioned in the Javadoc for `FilePathUtils.getNodeNameOrNull`.

Maybe time to go ahead and submit a PR to core to move `FilePathUtils` there for future use.

Reply:
No problem, will do

Reply:
`FilePath.toComputer` is `@CheckForNull`. Why isn't SpotBugs giving you an error on `FilePath.toComputer().getNode()`? Maybe try running `mvn spotbugs:check` locally to see if that gives you any warnings?

Reply:
Ah, the problem is that the POM makes FindBugs warnings non-fatal: https://github.com/jenkinsci/durable-task-plugin/blob/93b86c92fc0545cb4df79abe0ef2825a2c149827/pom.xml#L27

Locally, I would check for SpotBugs issues and fix all the warnings in code you modified. If that covers everything, I would delete that line from the configuration. If not, I guess I would leave it there for now, unless there are just one or two easy-to-fix warnings.

Reply:
Ok, and like I mentioned yesterday, I was concerned that throwing an `Exception` if any part of `FilePath.toComputer().getNode()` was null was too strict, but as far as I understand in this case they are either non-null or the script will fail to start later on when we use the `Launcher` anyway, so it shouldn't matter.